server/SwipeDecision.py

```diff
@@ -15,16 +16,66 @@ def getPotentialMatchList(currentUserId):
             cursor.execute(PotentialMatchQuery, userID)
             potentialListId = [i[0] for i in cursor.fetchall()]
 
+            newPotentialMatchList = []
+
+            if(len(potentialListId) != 0):
```

Comment:
if length of `potentialListId` id equals 0 why would you set `newPotentialMatchList = potentialListId `.  Both lists are already `[]` so the else statement for this doesn't do anything. You can remove it

Reply:
Put all the code within this if statement into a method called `getMatchesWithinMaxDistance` and add `newPotentialMatchList = getMatchesWithinMaxDistance(...)` within this if statement